### PR TITLE
Fix/windows case sensitivity

### PR DESCRIPTION
**Pull request by ph**
Fix uninstall problem on windows with case sensitivity

This PR fix an issue when trying to uninstall the Elastic Agent on
Windows by using a path that doesn't match the system case sensitivity.

This make sure the following are considered to be the same.

- C:/path-to-agent/elastic-agent uninstall
- c:/path-to-agent/elastic-agent uninstall

Fixes: #22268

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Review by blakerouse — COMMENTED** [review]
Have you been able to test this? I think there might be more places where case-sensitivity might be causing issues on Windows.

File: x-pack/elastic-agent/pkg/agent/install/installed.go [hunk 1]
```diff
@@ -58,6 +60,15 @@ func RunningInstalled() bool {
 		execDir = filepath.Dir(filepath.Dir(execDir))
 		execPath = filepath.Join(execDir, execName)
 	}
+
+	// Windows filesystem is case insensitive and the following paths are the same.
+	// - C:\TMP
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/install/installed.go:65]: maybe we can extract this to paths package and compare paths using paths.AreEqual or something similar

**Comment by ph** [comment]
@blakerouse I just tested the workflow referenced in the issue, nothing else I can do a  bit more of testing.

**Comment by gabriellandau** [comment]
> I just tested the workflow referenced in the issue

Since this at least fixes the user-reported bug #22268, can we merge it for now and open another issue later if other bugs come up?

**Pull request closed without merging**